lib/models/watcher.js

```diff
@@ -55,22 +59,19 @@ module.exports = class Watcher extends CoreObject {
     }
   }
 
-  _setupBroccoliWatcherBuild() {
-    let heimdallNode;
-
-    this.watcher.currentBuild
-      .then(hash => {
-        heimdallNode = hash.graph.__heimdall__;
-        return hash;
-      })
-      .finally(() => {
-        // guard against `build` rejecting
-        if (heimdallNode) {
-          // remove the heimdall subtree for this build so we don't leak.  If
-          // BROCCOLI_VIZ=1 then we have already output the json in `verboseOutput`.
-          heimdallNode.remove();
-        }
-      });
+  async _setupBroccoliWatcherBuild() {
+    try {
+      const hash = await this.watcher.currentBuild;
+      hash.graph.__heimdall__.remove();
+    } catch (e) {
+      if (e !== null && typeof e === 'object' && e.isBuilderError === true) {
+        // it is safe to ignore build errors here, as this code only exists to
+        // prune heimdall data after a build.
+      } else {
+        // this error is unexpected
+        throw e;
```

Comment:
IIRC re-throwing like this will not add `_setupBroccoliWatcherBuild` to the stack trace. Would it be useful for debugging purposes to know that this frame was involved in the failure?

Reply:
> Would it be useful for debugging purposes to know that this frame was involved in the failure?

Not particularly, especially considering the complexity of stitching the stack frames together.